### PR TITLE
Change default provider to Ember

### PR DESCRIPTION
**Pull request by rossf7**
Towards #7 

Adds support for the Ember provider and makes it the default.

```
go run .

curl -s http://localhost:8000/metrics | grep grid | head
# HELP grid_intensity_carbon_actual Actual carbon intensity for the electricity grid in this region.
# TYPE grid_intensity_carbon_actual gauge
grid_intensity_carbon_actual{provider="ember-climate.org",region="ARG"} 365.292
grid_intensity_carbon_actual{provider="ember-climate.org",region="ARM"} 206.522
grid_intensity_carbon_actual{provider="ember-climate.org",region="AUS"} 526.876
grid_intensity_carbon_actual{provider="ember-climate.org",region="AUT"} 145.083
grid_intensity_carbon_actual{provider="ember-climate.org",region="AZE"} 536.585
grid_intensity_carbon_actual{provider="ember-climate.org",region="BDI"} 275.862
grid_intensity_carbon_actual{provider="ember-climate.org",region="BEL"} 156.063
grid_intensity_carbon_actual{provider="ember-climate.org",region="BGD"} 559.606
```

Or for a single country

```
GRID_INTENSITY_REGION=ESP go run .

curl -s http://localhost:8000/metrics | grep grid | head
# HELP grid_intensity_carbon_actual Actual carbon intensity for the electricity grid in this region.
# TYPE grid_intensity_carbon_actual gauge
grid_intensity_carbon_actual{provider="ember-climate.org",region="ESP"} 193.737
```

Or using the carbonintensity.org.uk provider

```
GRID_INTENSITY_PROVIDER=carbonintensity.org.uk go run .

curl -s http://localhost:8000/metrics | grep grid | head
# HELP grid_intensity_carbon_actual Actual carbon intensity for the electricity grid in this region.
# TYPE grid_intensity_carbon_actual gauge
grid_intensity_carbon_actual{provider="carbonintensity.org.uk",region="UK"} 197
```

### CODE REVIEW

File: .github/workflows/kubernetes-integration-test.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
           tags: thegreenwebfoundation/grid-intensity-exporter:integration-test
       - 
         name: Create kubernetes Kind Cluster
-        uses: helm/kind-action@v1.1.0
+        uses: helm/kind-action@v1.3.0
```
Comment (rossf7) [inline .github/workflows/kubernetes-integration-test.yaml:28]: The integration test was failing. 

This upgrades the Kubernetes version we test with from 1.19 to 1.24 and means the test will pass 🤷‍♂️

File: go.mod [hunk 2]
```diff
@@ -1,9 +1,21 @@
 module github.com/thegreenwebfoundation/grid-intensity-exporter
 
-go 1.15
+go 1.17
 
 require (
 	github.com/cenkalti/backoff/v4 v4.1.0
 	github.com/prometheus/client_golang v1.8.0
-	github.com/thegreenwebfoundation/grid-intensity-go v0.0.0-20201120145608-ed4c07f2660c
+	github.com/thegreenwebfoundation/grid-intensity-go v0.1.1-0.20220622112548-6b75d66a9da8
```
Comment (rossf7) [inline go.mod:8]: Needs https://github.com/thegreenwebfoundation/grid-intensity-go/pull/24 to be merged and released

File: helm/grid-intensity-exporter/templates/configmap.yaml [hunk 3]
```diff
@@ -6,4 +6,6 @@ metadata:
     app.kubernetes.io/name: {{ .Release.Name }}
 data:
   gridIntensityProvider: {{ .Values.gridIntensity.provider | quote }}
+  {{ if .Values.gridIntensity.region }}
   gridIntensityRegion: {{ .Values.gridIntensity.region | quote }}
+  {{ end }}
```
Comment (rossf7) [inline helm/grid-intensity-exporter/templates/configmap.yaml:11]: Region is now optional so we need this extra templating.

File: main.go [hunk 4]
```diff
@@ -81,18 +88,35 @@ func NewExporter(provider, region string) (*Exporter, error) {
 func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	ctx := context.Background()
 
-	actualIntensity, err := e.apiClient.GetCarbonIntensity(ctx, e.region)
-	if err != nil {
-		log.Printf("failed to get carbon intensity %#v", err)
-	}
+	if e.region == "" {
```
Comment (rossf7) [inline main.go:91]: This is the new logic to return all countries if a region is not provided.

**Review by mrchrisadams — APPROVED** [review]
I have one query about whether regions should be optional, but I'm prepared to follow the use cases you have in mind here Ross, and otherwise I'm happy for this to be merged in. Danke!

**Comment by rossf7** [comment]
> Do you think it's better to just have the exporter list a single country?

> The data from ember is very unlikely to change, and I can't think of a use case where you would be listing all the countries for prometheus to consume, because previously the library is been a bit like allowing a computer to say:

Good point, I'll remove the changes listing all countries. I also don't have a good use case for it right now.

My thinking was about the dev experience. So the user wouldn't need to provide any configuration and could later select the country they are interested in.

But I think it's better and simpler 😅 to provide example install commands with a country and the user can change to their value.

**Pull request merged**